### PR TITLE
setPedStat for client side peds

### PR DESCRIPTION
**Pull request by ArranTuna**

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 1]
```diff
@@ -1811,6 +1813,50 @@ int CLuaPedDefs::SetPedAimTarget ( lua_State* luaVM )
...
+                argStream.SetCustomError ( "Stat must be 0 to 343 and value must be 0 to 1000." );
+               
+        // Dont let them set visual stats if they don't have the CJ model
+        if ( !argStream.HasErrors ( ) )
+            if ( ( usStat == 21 /* FAT */ || usStat == 23 /* BODY_MUSCLE */ ) && pPed->GetModel ( ) != 0 )
+                argStream.SetCustomError( "Fat and muscle stat can only be set on CJ skin." );
+
+        if ( !argStream.HasErrors ( ) )
+        {
+            pPed->SetStat ( usStat, fValue );
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: This should also change the stats for child peds.

See [the server side implementation](https://github.com/multitheftauto/mtasa-blue/blob/master/Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L3838)

You may need to create a similar method with [these lines](https://github.com/multitheftauto/mtasa-blue/blob/master/Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L3860-L3878).

You will need to add the new method [around here](https://github.com/multitheftauto/mtasa-blue/blob/master/Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L2170-L2185) (you can refer to the linked method as a simple example - you need not create an extra method checking for validity, but please do check for the CJ skin). Don't forget to update the header file too.

Ideally both the client/server side functions should behave exactly the same (with the only difference being that the client side function only works on client created peds).

**Review by qaisjp — DISMISSED** [review]
Other than the issue regarding propagation to child elements, your pull request looks good to me.

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 2]
```diff
@@ -1811,6 +1813,43 @@ int CLuaPedDefs::SetPedAimTarget ( lua_State* luaVM )
...
+    // Verify the argument
+    CClientEntity* pEntity = NULL;
+    unsigned short usStat = 0;
+    float fValue = 0;
+    CScriptArgReader argStream ( luaVM );
+    argStream.ReadUserData ( pEntity );
+    argStream.ReadNumber ( usStat );
+    argStream.ReadNumber ( fValue );
+
+    if ( !argStream.HasErrors ( ) )
```
Comment (sbx320) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp:1827]: This shouldn't check whether the argstream has an error 4 times in a row. ~~Putting all of it into one if-else-statement would improve readability.~~

**edit** Actually just removing the if in line 1834 should be fine, missed the SetCustomError at first. 

Also valid ped stats are 0 to ```NUM_PLAYER_STATS - 1``` according to the server implementation. Your PR accepts the stat ```NUM_PLAYER_STATS``` as well. 
Also ```usStat``` can never be less than zero as it's an unsigned number.

**Comment by qaisjp** [comment]
Thank you!

**Pull request merged**